<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by
Stephen Brawner (brawner@gmail.com)
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->

<robot
    name="plane_part" xmlns:xacro="http://wiki.ros.org/xacro">

    <!-- include the xacro files-->
    <xacro:include filename="$(find plane_part)/urdf/agv_moving_part.gazebo.xacro" />
    <xacro:include filename="$(find plane_part)/urdf/materials.xacro" />

    <xacro:property name="car_mass" value="20.0" />
    <xacro:property name="car_offset" value="0.23" />
    <xacro:property name="PI" value="3.1415926536" />

    <link name="agv_Link">
        <visual>
            <geometry>
                <mesh filename="package://plane_part/meshes/agv2_Link.dae" />
            </geometry>
            <material name="green" />
            <origin rpy="0.0 0.0 0.0" xyz="0 0 0" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://plane_part/meshes/agv2_Link.dae" />
            </geometry>
            <origin rpy="0.0 0.0 0.0" xyz="0 0 0" />
        </collision>
        <inertial>
            <mass value="${car_mass}" />
            <inertia ixx="${car_mass}" ixy="0.0" ixz="0.0" iyy="${car_mass}" iyz="0.0"
                izz="${car_mass}" />
        </inertial>
    </link>

    <joint
        name="agv_Joint"
        type="fixed">
        <origin
            xyz="0.0 0.0 0.144"
            rpy="0 0 0" />
        <parent
            link="agv_Link" />
        <child
            link="support_Link" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link
        name="support_Link">
        <inertial>
            <origin
                xyz="0.071702 1.7942E-05 0.0058498"
                rpy="0 0 0" />
            <mass
                value="91.018" />
            <inertia
                ixx="6.4041"
                ixy="-5.4882E-05"
                ixz="-3.6045E-06"
                iyy="6.3339"
                iyz="-1.8817E-05"
                izz="12.526" />
        </inertial>
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <mesh
                    filename="package://plane_part/meshes/support_Link.dae" />
            </geometry>
            <material
                name="">
                <color
                    rgba="0 1 0 1" />
            </material>
        </visual>
        <collision>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <mesh
                    filename="package://plane_part/meshes/support_Link.dae" />
            </geometry>
        </collision>
    </link>

    <joint
        name="support_Joint"
        type="fixed">
        <origin
            xyz="0.27 0.0 1.197"
            rpy="0 0 0" />
        <parent
            link="support_Link" />
        <child
            link="reference_Link" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link
        name="reference_Link">
        <inertial>
            <origin
                xyz="-0.36224 -7.2789E-07 -0.21599"
                rpy="0 0 0" />
            <mass
                value="10.768" />
            <inertia
                ixx="0.43054"
                ixy="-4.8572E-17"
                ixz="3.0383E-17"
                iyy="0.17237"
                iyz="2.6573E-17"
                izz="0.51657" />
        </inertial>
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <mesh
                    filename="package://plane_part/meshes/reference_Link.STL" />
            </geometry>
            <material
                name="">
                <color
                    rgba="0 1 1 1" />
            </material>
        </visual>
        <collision>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <mesh
                    filename="package://plane_part/meshes/reference_Link.STL" />
            </geometry>
        </collision>
    </link>

    <joint
        name="target1_Joint"
        type="fixed">
        <origin
            xyz="-0.67 0 0.075"
            rpy="0 0 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target1" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target1">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <mesh
                    filename="package://plane_part/meshes/aruco_marker.dae" scale="1 5 5" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target2_Joint"
        type="fixed">
        <origin
            xyz="-0.67 0.515 -0.015"
            rpy="0 ${PI/2} 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target2" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target2">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.01" length="0.002" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target3_Joint"
        type="fixed">
        <origin
            xyz="-0.67 -0.515 -0.015"
            rpy="0 ${PI/2} 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target3" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target3">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.01" length="0.002" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target4_Joint"
        type="fixed">
        <origin
            xyz="-0.67 0 -0.545"
            rpy="0 ${PI/2} 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target4" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target4">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.01" length="0.002" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target5_Joint"
        type="fixed">
        <origin
            xyz="-0.67 0 0.255"
            rpy="0 0 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target5" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target5">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <mesh
                    filename="package://plane_part/meshes/aruco_marker2.dae" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target6_Joint"
        type="fixed">
        <origin
            xyz="0.0 0.515 -0.045"
            rpy="0 ${PI/2} 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target6" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target6">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.01" length="0.002" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target7_Joint"
        type="fixed">
        <origin
            xyz="0.0 0.515 -0.515"
            rpy="0 ${PI/2} 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target7" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target7">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.01" length="0.002" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target8_Joint"
        type="fixed">
        <origin
            xyz="0.0 -0.515 -0.045"
            rpy="0 ${PI/2} 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target8" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target8">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.01" length="0.002" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

    <joint
        name="target9_Joint"
        type="fixed">
        <origin
            xyz="0.0 -0.515 -0.515"
            rpy="0 ${PI/2} 0" />
        <parent
            link="reference_Link" />
        <child
            link="image_target9" />
        <axis
            xyz="1 0 0" />
    </joint>

    <link name="image_target9">
        <visual>
            <origin
                xyz="0 0 0"
                rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.01" length="0.002" />
            </geometry>
            <material
                name="">
                <color
                    rgba="1 1 1 1" />
            </material>
        </visual>
    </link>

</robot>